### PR TITLE
Add test for cat error message when `ipfs init` hasn't been run

### PR DESCRIPTION
**Pull request by rht**
#1060

### CODE REVIEW

**Comment by whyrusleeping** [comment]
test failure is.... iptb. i really wish i could repro that locally, but i think its a dht issue.

This all LGTM

File: repo/fsrepo/fsrepo.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ type NoRepoError struct {
 var _ error = NoRepoError{}
 
 func (err NoRepoError) Error() string {
-	return fmt.Sprintf("no ipfs repo found in '%s'. please run: ipfs init ", err.Path)
+	return fmt.Sprintf("no ipfs repo found in %s. please run: ipfs init", err.Path)
```
Comment (jbenet) [inline repo/fsrepo/fsrepo.go]: maybe let's use a newline: 

```go
"no ipfs repo found in: %s.\nplease run: ipfs init"
```

File: test/sharness/t0020-init.sh [hunk 2]
```diff
@@ -21,9 +21,22 @@ test_expect_success "ipfs init fails" '
...
 
+# test no repo error message
+# this applies to `ipfs add sth`, `ipfs refs <hash>`
+test_expect_success "ipfs cat fails" '
+    export IPFS_PATH="$(pwd)/.ipfs" &&
+    test_must_fail ipfs cat Qmaa4Rw81a3a1VEx4LxB7HADUAXvZFhCoRdBzsMZyZmqHD 2> cat_fail_out
+'
+
+test_expect_success "ipfs cat no repo message looks good" '
+    echo "Error: no ipfs repo found in $IPFS_PATH. please run: ipfs init" > cat_fail_exp &&
```
Comment (jbenet) [inline test/sharness/t0020-init.sh]: will need to change it here too

**Comment by jbenet** [comment]
@rht this LGTM, two comments:

- [ ] add newline to error message
- [ ] rebase on master

**Comment by rht** [comment]
Rebased && Add newline.

Hey what is the short-hand notation for "except for the known breakages and/or iptb, test works"?

**Pull request merged**